haproxy/tests/with_mock.bats

```diff
@@ -0,0 +1,29 @@
+setup() {
+	load 'test_helper/bats-support/load' # this is required by bats-assert
+	load 'test_helper/bats-assert/load'
+
+	set -eu -o pipefail
+}
+
+[[ -n "${DEBUG_BATS:-}" ]] && set -x
+
+@test "should respond before timeout" {
+	run timeout 15 bash -c "until curl -fksm5 http://127.0.0.1:5984; do sleep 0.1; done"
+	assert_success
+}
+
+@test "should receive response from couchdb" {
+	run bash -c "curl -fksm5 http://127.0.0.1:5984/ | jq .couchdb"
+	assert_success
+	assert_output '"Welcome"'
+
+	run curl -Ifksm5 http://127.0.0.1:5984/doesnotexist
+	assert_failure
+	assert_line --partial --index 0 'HTTP/1.1 404 Not Found'
```

Comment:
I think this would be better as a separate test.

Reply:
Agreed, updated